aws/sdk-codegen/src/test/kotlin/software/amazon/smithy/rustsdk/SmokeTestsDecoratorTest.kt

```diff
@@ -109,7 +227,7 @@ class SmokeTestsDecoratorTest {
             // setting enabled by default in `.crate/config.toml` in test workspaces.
             // To work around this issue, we unset `RUSTFLAGS` to allow unused and dead code.
```

Comment:
As discussed offline, can we add a clarifying comment something along the lines of

> to perform a compilation-only test, we don't need to set `mapOf(Pair("RUSTFLAGS", "--cfg smoketests"))`?

I think it's worth clarifying.

Reply:
Added clarifying comment in https://github.com/smithy-lang/smithy-rs/pull/3840/commits/c20b7b8d1d3f6eeb0ad03e4ccfc4420eff777413